paasta_tools/secret_providers/vault.py

```diff
@@ -155,6 +155,14 @@ def get_secret_signature_from_data(self, data: Mapping[str, Any]) -> Optional[st
         else:
             return None
 
+    def get_secrets_from_vault_path(self, path=""):
+        data = self.clients[self.ecosystems[0]].read(path)
+        # returns None on empty path. return {} instead
```

Comment:
will we ever be passed in an empty path? and why is returning an empty dict preferable over None in this case?

Reply:
`None` would break the json serialization for the base64 dump me thinks. better to just return {} instead since if the secret is empty, it's not as catastrophic as if the secrets_sync script breaks. {} just gives us that serialization and still tells us it's empty at the same time. but open to suggestions

empty path shouldn't be passed for our use case but technically hvac.read doesn't care either given it's array of supported use cases https://hvac.readthedocs.io/en/stable/source/hvac_v1.html#hvac.v1.Client.read

what do you think is best?

Reply:
imo, I'd return `None` -> document that this function will return `None` if the secret doesn't exist -> check for None wherever we call this function

afaict, we could technically have an empty secret, but the current code would mean there's no way to distinguish between an empty secret and a key that doesn't exist

Reply:
++ makes sense to me, will make the change